build_tools/benchmarks/run_benchmarks_on_android.py

```diff
@@ -297,95 +306,132 @@ def run_benchmarks_for_category(
   normal_benchmark_tool_path = adb_push_to_tmp_dir(normal_benchmark_tool,
                                                    relative_dir="normal-tools",
                                                    verbose=verbose)
+  benchmark_results_dir = os.path.join(tmp_dir, "benchmark-results")
+  os.makedirs(benchmark_results_dir, exist_ok=True)
+
+  captures_dir = os.path.join(tmp_dir, "captures")
   if traced_benchmark_tool is not None:
+    os.makedirs(captures_dir, exist_ok=True)
     traced_benchmark_tool_path = adb_push_to_tmp_dir(
         traced_benchmark_tool, relative_dir="traced-tools", verbose=verbose)
 
   results = []
+  errors = []
+  previous_benchmarks = previous_benchmarks if previous_benchmarks else set()
+  previous_captures = previous_captures if previous_captures else set()
 
   # Push all model artifacts to the device and run them.
   root_benchmark_dir = os.path.dirname(benchmark_category_dir)
   for benchmark_case_dir in benchmark_case_dirs:
     benchmark_info = compose_benchmark_info_object(device_info,
                                                    benchmark_category_dir,
                                                    benchmark_case_dir)
+    benchmark_key = str(benchmark_info)
+    if benchmark_key in previous_benchmarks and (not do_capture or benchmark_key
+                                                 in previous_captures):
+      continue
     print(f"--> benchmark: {benchmark_info} <--")
+    try:
```

Comment:
Comment: "Try benchmarking and performing captures on the Android device now. This ignores previous benchmarks and captures if they exist and keeps going if requested".